### PR TITLE
Add macromolecular complex compendium

### PR DESCRIPTION
**Pull request by Shalsh23**

### CODE REVIEW

File: src/createcompendia/macromolecular_complex.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+       :identifiers: a list of files from which to read identifiers and optional categories"""
+    dicts = {}
+    types = {}
+    uniques = [COMPLEXPORTAL]
+    for ifile in identifiers:
+        print('loading ', ifile)
+        new_identifiers, new_types = read_identifier_file(ifile)
+        glom(dicts, new_identifiers, unique_prefixes=uniques)
+        types.update(new_types)
+    sets = set([frozenset(x) for x in dicts.values()])
```
Comment (Shalsh23) [inline src/createcompendia/macromolecular_complex.py:18]: @cbizon I am not sure if these couple of lines are correct.

**Comment by Shalsh23** [comment]
The `build_compendia` part needs review and the code overall needs testing before being ready to merge.

**Comment by Shalsh23** [comment]
Fixes #54

**Comment by gaurav** [comment]
- [ ] @gaurav to test and report back
- [ ] @cbizon to review

**Comment by cbizon** [comment]
Yes, the writer will ignore any identifier with a prefix that is undefined in the biolink model.  I thought we had got that added, but I just double checked and apparently we didn't.  So I think we want to make a PR to add it to biolink as an allowed prefix for MacromolecularComplexMixin. 

I think we shouldn't let this delay getting out the new version though.  The only users are the yeastkop folks, and they can work around nn misses for the time being.

**Comment by gaurav** [comment]
Aha! I ran into the same problem with PR #50, where I'm trying to add UMLS concepts to the [OrganismTaxon](https://biolink.github.io/biolink-model/docs/OrganismTaxon.html) compendia, which currently only recognizes NCBITaxon and MESH. I was able to make it work by adding an [`extra_prefixes` argument to `write_compendium`](https://github.com/TranslatorSRI/Babel/blob/9a029a1228efb5eba6e3ba35d8e6a2e8138384ab/src/babel_utils.py#L202-L211), which allows me to force it to include a prefix that isn't yet included in the Biolink model. With that change, the UMLS taxa appear to be included -- I accidentally deleted a bunch of Babel outputs, so I'm re-running the process now, and should have an updated list of leftover UMLS terms for you later today. Do you think something like that would be a good short-term fix for this issue in this PR?

**Comment by gaurav** [comment]
Next steps:
- [x] Request prefix: https://github.com/biolink/biolink-model/issues/1088
- [x] Confirmed that adding COMPLEXPORTAL as an extra prefix is sufficient for generating the compendium.
- [ ] Question: since we don't have any cross references between ComplexPortal and anything else, we're currently only generating single identifier clusters for MacromolecularComplexMixins -- I assume that's okay, but just wanted to double-check!
- [ ] Question: we currently don't include labels or synonyms for MacromolecularComplexMixins -- should I add those in?
- [ ] Question: we currently incorporate _S. cerevisiae_ complexes only ([relevant code line](https://github.com/TranslatorSRI/Babel/blob/b0d638e4d8532b5103bfbaa746665f90cb90fadf/src/datahandlers/complexportal.py#L5)) -- should we extend this to the other 27 species available at http://ftp.ebi.ac.uk/pub/databases/intact/complex/current/complextab/?

File: src/datahandlers/complexportal.py [hunk 2]
```diff
@@ -12,7 +12,7 @@ def make_labels_and_synonyms(infile, labelfile, synfile):
             sline = line.split("\t")
             id = sline[0]
             label = sline[1] # recommended name
-            outl.write(f'Complex ac:{id}\t{label}\n')
+            outl.write(f'ComplexPortal:{id}\t{label}\n')
```
[the remark this answers is not in the document]
Comment (gaurav) [reply src/datahandlers/complexportal.py]: Fixed in e838712.

**Review by cbizon — APPROVED** [review]

**Pull request merged**